src/Type/Php/ArrayColumnFunctionReturnTypeExtension.php

```diff
@@ -180,4 +183,19 @@ private function getOffsetOrProperty(Type $type, Type $offsetOrProperty, Scope $
 		return TypeCombinator::union(...$returnTypes);
 	}
 
+	private function castToArrayKeyType(Type $type): Type
+	{
+		$isArray = $type->isArray();
+		if ($isArray->yes()) {
+			return new IntegerType();
```

Comment:
I don't get it. Array gets cast to integer?

Reply:
See https://sandbox.onlinephpfunctions.com/code/381d0182b85cc698b02b18b82cd9dae0da3f7a00
I intentionally used 'foo' instead of 'type' in the array_column call in this example. So this seems to be the behaviour.

Reply:
Yes exactly. In PHP < 8, whenever a value cannot be used as an array key (either because it is an array, or the value is missing entirely), the next available integer array key is used (as if you'd do `$arr[] = $value`). In PHP >= 8, using an array value as key causes a fatal error, which of course is an opportunity for PHPStan to detect in the future.